### PR TITLE
Feature/enable client db tests

### PR DESCRIPTION
**Pull request by mateuszaaa**
Unifies way how block are handled - introduces exceptions for handling first produced block in pallet_timestamp

### CODE REVIEW

**Comment by mateuszaaa** [comment]
proof for compatibility with e2e tests https://github.com/mangata-finance/mangata-node/pull/124

File: frame/timestamp/src/lib.rs [hunk 1]
```diff
@@ -167,35 +167,19 @@ decl_module! {
...
-
-            <T::OnTimestampSet as OnTimestampSet<_>>::on_timestamp_set(timestamp);
-        }
-
+		fn set(origin, #[compact] now: T::Moment) {
+			ensure_none(origin)?;
+			assert!(!<Self as Store>::DidUpdate::exists(), "Timestamp must be updated only once in the block");
+			let prev = Self::now();
+			assert!(
+				prev.is_zero() || now >= prev + T::MinimumPeriod::get(),
```
Comment (gleb-urvanov) [inline frame/timestamp/src/lib.rs:175]: If `let prev = Self::now();`, how can `now >= prev + T::MinimumPeriod::get()` be true?

**Review by gleb-urvanov — CHANGES_REQUESTED** [review]
Was it tested with several nodes running?

File: frame/timestamp/src/lib.rs (hunk 1 above)
Comment (mateuszaaa) [reply frame/timestamp/src/lib.rs:175]: Actually i just reverted origin implementation(tags/v2.0.1) but i believe the first condition is for handling first block and the second one is for all the others

**Review by gleb-urvanov — APPROVED** [review]

**Pull request merged**